external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaSourceProviderSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.kafka010
+
+import java.util.Locale
+
+import scala.collection.JavaConverters._
+
+import org.scalatest.PrivateMethodTester
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.sources.v2.reader.Scan
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+class KafkaSourceProviderSuite extends SparkFunSuite with PrivateMethodTester {
+  private val pollTimeoutMsMethod = PrivateMethod[Long]('pollTimeoutMs)
+
+  test("SPARK-28142 - continuous mode - options should be handled as case-insensitive") {
+    def getPollTimeoutMsFromContinuousStream(options: CaseInsensitiveStringMap): Long = {
```

Comment:
Is it possible to move this into `KafkaSourceProviderSuite` and generalize it (for instance add private method as param an T return type)?

Reply:
For instance I would like to reach `KafkaOffsetReader` inside `KafkaContinuousStream`. I need something like this but would be good to generalize it a bit in advance:
```
    def getKafkaOffsetReaderFromContinuousStream(options: CaseInsensitiveStringMap): KafkaOffsetReader = {
      val scan = getKafkaDataSourceScan(options)
      val stream = scan.toContinuousStream("dummy").asInstanceOf[KafkaContinuousStream]
      stream.invokePrivate(offsetReader())
    }
```

Reply:
The other possibility is to split creation and invokePrivate...

Reply:
Nice suggestion! I guess you would need to do similar one if this patch doesn't provide generic one, so welcome suggestion on the way to reuse the code.